### PR TITLE
Add gitgutter with flycheck

### PR DESCRIPTION
**Pull request by Robdel12**
Thanks @wwilsman 😈

### CODE REVIEW

File: frontmacs-editing.el [hunk 1]
```diff
@@ -27,6 +28,13 @@
 ;; undo visual tree
 (global-undo-tree-mode 1)
 
+;; setup flycheck
+(setq flycheck-indication-mode 'right-fringe)
```
Comment (cowboyd) [inline frontmacs-editing.el]: We should be more descriptive with our descriptions. The key here is to ask "could someone follow along here?" If there's a relevant link, it's good to include it.

File: frontmacs-editing.el [hunk 2]
```diff
@@ -27,6 +28,13 @@
...
+(define-fringe-bitmap 'flycheck-fringe-bitmap-double-arrow
+  [0 0 0 0 0 4 12 28 60 124 252 124 60 28 12 4 0 0 0 0])
+
```
Comment (cowboyd) [inline frontmacs-editing.el]: This definitely needs unpacking :)

**Comment by wwilsman** [comment]
I actually removed git gutter to use diff-hl. 

git gutter had issues with autosaving and diff-hl has a live mode

File: frontmacs-vcs.el [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
+  nil nil 'center)
+(define-fringe-bitmap 'git-gutter-fr:modified
+  [224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224 224]
+  nil nil 'center)
+(define-fringe-bitmap 'git-gutter-fr:deleted
+  [0 0 0 0 0 0 0 0 0 0 0 0 0 128 192 224 240 248]
+  nil nil 'center)
+
+(global-git-gutter-mode +1)
+(add-hook `prog-mode-hook `git-gutter-mode)
```
Comment (cowboyd) [inline frontmacs-vcs.el]: I think there might be a mis-match.

1) there are modes that aren't programming modes (markdown-mode) that should have git gutter
2) there are programming modes that exist outside a git repo that shouldn't

What if we add a hook to creating the buffer that checks if there is a git repo, and then adds it
shouldn't this be only added to buffers inside

**Review by cowboyd — CHANGES_REQUESTED** [review]

**Review by cowboyd — APPROVED** [review]

**Pull request merged**